### ISSUE TITLE
`paths.base` should automatically add the pase path to relative links

### ISSUE THREAD

**Issue by DetachHead** [sveltejs/kit#8084]
### Describe the problem

https://kit.svelte.dev/docs/configuration#paths
> Note that you need to prepend all your root-relative links with the base value or they will point to the root of your domain, not your base (this is how the browser works). You can use base from $app/paths for that: <a href="{base}/your-page">Link</a>. If you find yourself writing this often, it may make sense to extract this into a reusable component.

i don't get this, why can't sveltekit just add the base path automatically? to me it seems like this option is pointless if you still need to manually specify it everywhere

### Describe the proposed solution

automatically add the base path to relative links

### Alternatives considered

using `import { base } from '$app/paths'` but that doesn't work in `.html` files

### Importance

would make my life easier

**Comment by KotlinIsland** [comment]
> If you find yourself writing this often, it may make sense to extract this into a reusable component.

I'm sure that everyone that would ever use `paths.base` would use this absolutely everywhere. Could there be some kind of functionality that reduces the boilerplate for this feature?

**Comment by dummdidumm** [comment]
Automagically adding it sounds like a can of worms. If we do it at runtime, it will fall flat if client JS is disabled, because nothing can be prefixed then. So this would need to be a preprocessor or something similar that runs at build time. Then there's the question which links should get the base path treatment - all? What about dynamically added hrefs? Do there need to be exceptions for specific paths? This all sounds worse than creating a simple `A` component yourself and have full control.

**Comment by KotlinIsland** [comment]
Could SvelteKit automatically add a `<base>` if `paths.base` has been configured?

**Comment by DetachHead** [comment]
yeah i don't get why we need to add this boilerplate when that's what you'd want to do 99% of the time. why can't edge cases like dynamic hrefs have some syntax to opt out instead?

**Comment by DetachHead** [comment]
so i just found out that when using relative paths (`src="foo"`) it already behaves the way i want. it's only when prepending a `/` (`src="/foo"`) that you need to append the `base` (`src="{base}/foo"`).

as for the `<base>` tag, that seems to only work on paths that *don't* have the leading `/`, so it wouldn't work in this case

**Comment by KotlinIsland** [comment]
Hey, @Rich-Harris, using a `<base>` element is not fruitful, it doesn't affect non-relative paths (which are the only kind of paths that are relevant to `paths.base`) what if Svelte could do something like this with any element that takes a path automatically?

```svelte
<script lang="ts">
    import { base } from '$app/paths'

    export let href: string

    $: basedHref = href[0] === '/' ? `${base}${href}` : href
</script>

<a href={basedHref} {...$$restProps} />
```
